lms/templates/courseware/course_navigation.html

```diff
@@ -44,19 +44,23 @@
                     <%
                     tab_is_active = tab.tab_id in (active_page, default_tab)
                     %>
-                    <li class="nav-item ${'active' if tab_is_active else ''}">
-                        <a href="${tab.link_func(course, reverse)}" class="nav-link">
-                            ${_(tab.name)}
-                            % if tab_is_active:
-                                <span class="sr-only">, ${_('current location')}</span>
-                            %endif
-                            % if tab_image:
-                                ## Translators: 'needs attention' is an alternative string for the
-                                ## notification image that indicates the tab "needs attention".
-                                <img src="${tab_image}" alt="${_('needs attention')}" />
-                            %endif
-                        </a>
-                    </li>
+                        %  if masquerade is not UNDEFINED and tab.name == 'Instructor' and masquerade is not None and masquerade.role == 'student':
```

Comment:
nit: for ease of readability of this code I might order as

```
if masquerade.role == 'student' and tab.name == 'Instructor' and masquerade is not UNDEFINED and masquerade is not None:
```

Reply:
we should calculate these values on each iteration of the tab. so I would suggest pulling `masquerade` related conditions on to the top of the page (b/c they won't change for each tab). Once done, this can be simplified as

```
% tab.name == 'Instructor'  and show_Instructor == true:
    <li ....
    # rest of the code goes here. 
```